tools/distpartitioning/dataset_utils.py

```diff
@@ -33,46 +33,63 @@ def get_dataset(input_dir, graph_name, rank, num_node_weights):
+        node_feature_tids[ntype_name] = []
+        for feat_name, feat_data in ntype_feature_data.items():
+            assert len(feat_data) == world_size
+            my_feat_data = feat_data[rank]
+            if (os.path.isabs(my_feat_data[0])):
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(my_feat_data[0]))
+            else:
+                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(input_dir+my_feat_data[0]))
 
-    #done build node_features locally. 
-    if len(node_features) <= 0: 
-        print('[Rank: ', rank, '] This dataset does not have any node features')
-    else: 
-        for k, v in node_features.items():
-            print('[Rank: ', rank, '] node feature name: ', k, ', feature data shape: ', v.size())
+            v = [feat_name, my_feat_data[1], my_feat_data[2]]
+            l = node_feature_tids[ntype_name]
+            l.append(v)
+            node_feature_tids[ntype_name] = l
 
-    #read (split) xxx_nodes.txt file
-    node_file = input_dir+'/'+graph_name+'_nodes'+'_{:02d}.txt'.format(rank)
-    node_data = np.loadtxt(node_file, delimiter=' ', dtype='int64')
-    nodes_datadict = {}
-    nodes_datadict[constants.NTYPE_ID] = node_data[:,0]
-    type_idx = 0 + num_node_weights + 1
-    nodes_datadict[constants.GLOBAL_TYPE_NID] = node_data[:,type_idx]
-    print('[Rank: ', rank, '] Done reading node_data: ', len(nodes_datadict), nodes_datadict[constants.NTYPE_ID].shape)
+    #read my nodes for each node type
+    node_tids = {}
+    node_data = schema_map["nid"]
+    for ntype_name, ntype_info in node_data.items():
+        v = []
+        node_file_info = ntype_info["data"]
+        for idx in range(len(node_file_info)):
+            v.append((node_file_info[idx][1], node_file_info[idx][2]))
+        node_tids[ntype_name] = v
 
-    #read (split) xxx_edges.txt file
+    #read my edges for each edge type
+    edge_tids = {}
     edge_datadict = {}
-    edge_file = input_dir+'/'+graph_name+'_edges'+'_{:02d}.txt'.format(rank)
-    edge_data = np.loadtxt(edge_file, delimiter=' ', dtype='int64')
-    edge_datadict[constants.GLOBAL_SRC_ID] = edge_data[:,0]
-    edge_datadict[constants.GLOBAL_DST_ID] = edge_data[:,1]
-    edge_datadict[constants.GLOBAL_TYPE_EID] = edge_data[:,2]
-    edge_datadict[constants.ETYPE_ID] = edge_data[:,3]
+    edge_data = schema_map["eid"]
+    for etype_name, etype_info in edge_data.items():
+        if(etype_info["format"] == "csv"):
+            edge_info = etype_info["data"]
+            assert len(edge_info) == world_size
+
+            data = np.loadtxt(edge_info[rank][0], delimiter=' ', dtype=np.int64)
+            for idx, k in enumerate([constants.GLOBAL_SRC_ID, constants.GLOBAL_DST_ID, constants.GLOBAL_TYPE_EID, constants.ETYPE_ID]):
+                if k in edge_datadict:
+                    v = edge_datadict[k]
+                    v = np.concatenate((v, data[:,idx]))
```

Comment:
again, don't use concatenation in each iteration. it makes code long and inefficient.

Reply:
With the previous comment, this code has been rendered outdated. This code has been updated in response to previous comment.